### PR TITLE
implemented beer searching

### PR DESCRIPTION
**Pull request by kovpaka**

### CODE REVIEW

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import SearchInput, { SearchInputModes } from '../SearchInput/SearchInput.jsx';
+import './BeerApiComponent.scss';
+
+const beerApi = 'https://api.punkapi.com/v2/beers?per_page=5&beer_name=';
```
Comment (siarheikarabko) [inline src/Components/BeerApiComponent/BeerApiComponent.jsx:5]: beerApi**Url**

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            <p className="search-result-block__no-result-message">No result</p>
+        );
+    }
+
+    return (
+        <div className="beer-api-task">
+            <SearchInput
+                placeholder="Search"
+                mode={SearchInputModes.AfterEnter}
+                onSearch={onSearchHandler}
```
Comment (siarheikarabko) [inline src/Components/BeerApiComponent/BeerApiComponent.jsx:61]: As I understand, you do not display anything until a user types something in the search input. Consider showing beers just after the initial loading as well.

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import SearchInput, { SearchInputModes } from '../SearchInput/SearchInput.jsx';
+import './BeerApiComponent.scss';
+
+const beerApi = 'https://api.punkapi.com/v2/beers?per_page=5&beer_name=';
+
+export default function BeerApiComponent() {
+    const [beers, setBeer] = useState();
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:8]: Sure, fixed

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    let searchResultItems;
+    const onSearchHandler = (searchInput) => {
+        fetch(beerApi + searchInput)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:12]: Got it, fixed

File: src/Components/BeerApiComponent/BeerApiComponent.jsx (hunk 1 above)
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:5]: fixed.

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        beer_name: element.name,
+                        description: element.description,
+                        image: element.image_url,
+                    };
+
+                    beersArray.push(beer);
+                });
+
+                setBeer({
+                    beersArray,
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:29]: Done

File: src/components/SearchInput/SearchInput.jsx [hunk 6]
```diff
@@ -21,7 +21,7 @@ export default function SearchInput(props) {
     );
 
     const handleKeyDown = (e) => {
-        if (e.key === 'Enter') {
+        if (e.key === 'Enter' && e.target.value !== '') {
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/components/SearchInput/SearchInput.jsx:24]: Fixed

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+const beerApi = 'https://api.punkapi.com/v2/beers?per_page=5&beer_name=';
+
+export default function BeerApiComponent() {
+    const [beers, setBeer] = useState();
+
+    let searchResultItems;
+    const onSearchHandler = (searchInput) => {
+        fetch(beerApi + searchInput)
+            .then((response) => response.text())
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:13]: Done

[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:13]: Fone

File: src/Components/BeerApiComponent/BeerApiComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                </div>
+                <h3 className="search-result-block__title">{el.name}</h3>
+                <p className="search-result-block__description">
+                    {el.description}
+                </p>
+            </div>
+        ));
+    } else {
+        searchResultItems = (
+            <p className="search-result-block__no-result-message">No result</p>
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:52]: I don't know where I had to change it, so I changed it in all occurrences)

File: src/Components/BeerApiComponent/BeerApiComponent.jsx (hunk 2 above)
Comment (kovpaka) [reply src/Components/BeerApiComponent/BeerApiComponent.jsx:61]: Done.

File: src/components/BeerApiTask/BeerApiTask.jsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import SearchInput, { SearchInputModes } from '../SearchInput/SearchInput.jsx';
+import BeerList from '../BeerList/BeerList.jsx';
+import './BeerApiTask.scss';
+
+const beerApiUrl = 'https://api.punkapi.com/v2/beers?per_page=5';
+
+export default function BeerApiComponent() {
```
Comment (siarheikarabko) [inline src/components/BeerApiTask/BeerApiTask.jsx:8]: Please, make the component name to be consistent with the file. Moreover, the component class names aren't consistent with the component name...

File: src/components/BeerApiTask/BeerApiTask.jsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        setBeers(beersArray);
+    }
+
+    useEffect(() => {
+        onSearchHandler('');
+    }, []);
+
+    if (beers.length > 0) {
+        searchResultsItems = beers.map((beer) => (
+            <BeerList
```
Comment (siarheikarabko) [inline src/components/BeerApiTask/BeerApiTask.jsx:34]: It's more a beer list item rather beer list. Beer list should contain the logic of beer rendering 
`beers.map((beer) => <BeerListItem />`
Also, it's a good idea to move there the logic of rendering "no results" logic as well.
So, your BeerApiComponent (should be renamed) will be
```
        <div className="beer-api-task">
            <SearchInput
                placeholder="Search"
                mode={SearchInputModes.Immediate}
                onSearch={onSearchHandler}
            />
            <BeerList beers={beers} />
        </div>
```

File: src/components/BeerApiTask/BeerApiTask.scss [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    flex-direction: column;
+    justify-content: center;
+    padding: 0 15%;
+}
+
+.beer-api-task .beer-list-item {
+    margin-top: 20px;
+}
+
+.no-results-message {
```
Comment (siarheikarabko) [inline src/components/BeerApiTask/BeerApiTask.scss:12]: uhh.... one component - one BEM block...

File: src/components/BeerList/BeerList.jsx [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './BeerList.scss';
+
+export default function BeerList(props) {
```
Comment (siarheikarabko) [inline src/components/BeerList/BeerList.jsx:4]: The component name is BeerList, but it actually renders a beer list item....

File: src/components/BeerApiTask/BeerApiTask.jsx (hunk 9 above)
Comment (kovpaka) [reply src/components/BeerApiTask/BeerApiTask.jsx]: Fixed

File: src/components/BeerApiTask/BeerApiTask.jsx (hunk 10 above)
Comment (kovpaka) [reply src/components/BeerApiTask/BeerApiTask.jsx]: Fixed

File: src/components/BeerApiTask/BeerApiTask.scss (hunk 11 above)
Comment (kovpaka) [reply src/components/BeerApiTask/BeerApiTask.scss]: I'm really ,really sorry

File: src/components/BeerList/BeerList.jsx (hunk 12 above)
Comment (kovpaka) [reply src/components/BeerList/BeerList.jsx:5]: My bad, fixed

File: src/components/BeerApiTask/BeerApiTask.scss [hunk 13]
```diff
@@ -0,0 +1,10 @@
+.beer-api-task {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    padding: 0 15%;
+}
+
+.beer-api-task .beer-list-item {
```
Comment (siarheikarabko) [inline src/components/BeerApiTask/BeerApiTask.scss:8]: `BeerApiTask` shouldn't know anything about a beer list item and do not control list item view such a way.

File: src/components/BeerListItem/BeerListItem.jsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './BeerListItem.scss';
+
+export default function BeerListItem(props) {
+    return (
+        <li key={props.id} className="beer-list-item">
```
Comment (siarheikarabko) [inline src/components/BeerListItem/BeerListItem.jsx:6]: Please check, `key={props.id}` probably isn't needed here, it's enough to pass it as a prop to the component.

File: src/components/BeerApiTask/BeerApiTask.scss (hunk 13 above)
Comment (kovpaka) [reply src/components/BeerApiTask/BeerApiTask.scss:8]: Yes, sorry, I just didn't see this.

File: src/components/BeerListItem/BeerListItem.jsx (hunk 14 above)
Comment (kovpaka) [reply src/components/BeerListItem/BeerListItem.jsx:6]: yes, removed

**Pull request merged**